Source/WebCore/css/ShorthandSerializer.cpp

```diff
@@ -1343,16 +1343,23 @@ String ShorthandSerializer::serializeSingleAnimationRange(CSSValue& value, Singl
 
 String ShorthandSerializer::serializeAnimationRange() const
 {
-    auto& start = longhandValue(0);
-    RefPtr startPair = dynamicDowncast<CSSValuePair>(start);
-    auto startID = startPair ? startPair->first().valueID() : start.valueID();
-
-    auto serializedStart = serializeSingleAnimationRange(start, SingleTimelineRange::Type::Start);
-    auto serializedEnd = serializeSingleAnimationRange(longhandValue(1), SingleTimelineRange::Type::End, startID);
+    StringBuilder serializedString;
+    auto* startList = dynamicDowncast<CSSValueList>(longhandValue(0));
+    auto* endList = dynamicDowncast<CSSValueList>(longhandValue(1));
+    if (startList && endList) {
+        ASSERT(startList->size() == endList->size());
+        for (unsigned i = 0; i < startList->size(); i++) {
+            auto start = startList->item(i);
+            RefPtr startPair = dynamicDowncast<CSSValuePair>(start);
+            auto startID = startPair ? startPair->first().valueID() : start->valueID();
+
+            auto serializedStart = serializeSingleAnimationRange(*start, SingleTimelineRange::Type::Start);
+            auto serializedEnd = serializeSingleAnimationRange(*endList->item(i), SingleTimelineRange::Type::End, startID);
+            serializedString.append(serializedEnd.isEmpty() ? serializedStart : makeString(serializedStart, ' ', serializedEnd), (i < startList->size() - 1) ? ", "_s : emptyString());
+        }
```

Comment:
```suggestion
            serializedString.append(
                serializedEnd.isEmpty() ? serializedStart : makeString(serializedStart, ' ', serializedEnd),
                i == startList->size() - 1 ? ""_s : ", "_s
            );
        }
```

Can we format like this? It would be even better if you can use `interleave()` instead of this loop.

Reply:
Fixed. I don't think we can use interleave on a `CSSValueList`.